### PR TITLE
[Framework] Implement basic transaction fee logic

### PR DESCRIPTION
**Pull request by jolestar**
## Summary

Implement basic transaction fee logic. 

1. Auto-call faucet when a new account is created.
2. Deduct gas in the `post_execute` function.
3. Refactor coin, provide some extend function via `private_generics`.

part of #421 

TODO:

1. Should we keep the coin burn_cap, freeze_cap, mint_cap? Or provide the `extend` functions via `private_generics`?
2. Gas faucet #739

### CODE REVIEW

File: crates/rooch-framework/sources/coin.move [hunk 1]
```diff
@@ -335,6 +338,25 @@ module rooch_framework::coin {
         amount: u256,
     ): Coin<CoinType> {
         let addr = signer::address_of(account);
+        withdraw_interal<CoinType>(ctx, addr, amount) 
+    }
+
+    #[private_generics(CoinType)]
```
Comment (baichuan3) [inline crates/rooch-framework/sources/coin.move:344]: Also provide an extend function via private_generics for `initialize`?

File: crates/rooch-framework/sources/coin.move [hunk 2]
```diff
@@ -736,115 +778,4 @@ module rooch_framework::coin {
...
 
-    //
-    // Tests
-    //
-
-    #[test_only]
-    struct FakeCoin {}
-
-    #[test(source = @0xa0a, destination = @0xb0b, mod_account = @rooch_framework)]
-    public entry fun test_end_to_end(
```
Comment (baichuan3) [inline crates/rooch-framework/sources/coin.move:747]: Test cases in `coin.move` is all for entry funciton, move them into `coin_test.move`?

File: crates/rooch-framework/sources/coin.move (hunk 1 above)
Comment (baichuan3) [reply crates/rooch-framework/sources/coin.move:344]: Indeed, `register` will be more clarify then `initialize`

Comment (jolestar) [reply crates/rooch-framework/sources/coin.move:340]: Ok, I will refactor in future PR.

File: crates/rooch-framework/sources/coin.move (hunk 2 above)
Comment (jolestar) [reply crates/rooch-framework/sources/coin.move:747]: Done

**Review by baichuan3 — APPROVED** [review]

**Pull request merged**